modules/miner/miner.go

```diff
@@ -6,32 +6,27 @@ import (
 	"sync"
 	"time"
 
-	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/crypto"
 	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/types"
 )
 
 const (
-	iterationsPerAttempt = 16 * 1024
-
-	// headerForWorkMemory is the number of previous calls to 'headerForWork'
-	// that are remembered. Additionally, 'headerForWork' will only poll for a
-	// new block every 'headerForWorkMemory / blockForWorkMemory' times it is
+	// HeaderMemory is the number of previous calls to 'header'
+	// that are remembered. Additionally, 'header' will only poll for a
+	// new block every 'headerMemory / blockMemory' times it is
 	// called. This reduces the amount of memory used, but comes at the cost of
-	// not always having the most recent transactions
-	headerForWorkMemory = 1000
-
-	// blockForWorkMemory is the maximum number of blocks the miner will store
-	// Blocks take up to 2 megabytes of memory, so it is important to keep a cap
-	blockForWorkMemory = 20
-
-	// secondsBetweenBlocks is the maximum amount of time the block manager will
-	// go between generating new blocks. If the miner is not polling more than
-	// headerForWorkMemory / blockForWorkMemory blocks every secondsBetweenBlocks
-	// then the block manager will create new blocks in order to keep the miner
-	// mining on the most recent block, but this will come at the cost of preventing
-	// the block manger from storing as many headers
-	secondsBetweenBlocks = 30
+	// not always having the most recent transactions.
+	HeaderMemory = 10000
+
+	// BlockMemory is the maximum number of blocks the miner will store
+	// Blocks take up to 2 megabytes of memory, which is why this number is
+	// limited.
+	BlockMemory = 50
+
+	// MaxSourceBlockAge is the maximum amount of time that is allowed to
+	// elapse between generating source blocks.
+	MaxSourceBlockAge = 60 * time.Second
```

Comment:
hmm. Seems like a strange reason to me (why would an optimizer look at godoc? If they were serious enough to be optimizing specifically for this code, wouldn't they look at the source?) but whatevs